profiler/main.py

```diff
@@ -0,0 +1,169 @@
+    custom_out = [prof1, prof2]
+    data: ProfileData = ProfileData(
+        process=" ".join(process),
+        custom_out=custom_out,
+        rt=real_time,
+        ut=usr_time,
+        st=sys_time,
+        max_set_size=max_resident_set_size,
+        page_reclaims=page_reclaims,
+        page_faults=page_faults,
+        cycles_elapsed=cycles_elapsed,
+        peak_memory=peak_memory_footprint,
+        date=date.today(),
+        now=datetime.utcnow().strftime("%Y-%m-%d %H:%M:%S"),
+        tiledb_stats=tiledb_stats,
+        somacore_version=somacore.__version__,
+        tiledbsoma_version=tiledbsoma.__version__,
+    )
+    return data
+
+
+def main():
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "process", nargs="+", help="The main process and its arguments to run"
+    )
+    parser.add_argument(
+        "-p1",
+        "--prof1",
+        required=False,
+        help="Second profiler and its args to use (like python and R) to produce flamegraph of calls",
+    )
+    parser.add_argument(
+        "-p1out",
+        "--prof1_output",
+        required=False,
+        help="The flamegraph output produced by prof1",
+    )
+
+    parser.add_argument(
+        "-p2",
+        "--prof2",
+        required=False,
+        help="Third profiler and its args to use (like python and R) to produce flamegraph of calls",
+    )
+    parser.add_argument(
+        "-p2out",
+        "--prof2_output",
+        required=False,
+        help="The flamegraph output produced by prof2",
+    )
+    args = parser.parse_args()
+
+    print(f"Process to be run: {args.process}")
+    # Running the main process using time -v to get detailed memory and time"""
+    p = subprocess.Popen(
+        ["/usr/bin/time", "-al"] + args.process, stdout=PIPE, stderr=PIPE
+    )
+    print(f"Running main process, PID = {p.pid}")
+
+    # Running additional profilers to extract flame graphs for the run
+    p1 = None
+    p2 = None
+    if args.prof1 is not None:
+        if args.prof1_output is not None:
+            command = args.prof1.replace("PID", str(p.pid))
+            p1 = subprocess.Popen(command.split())
+            # potential example of a custom profiler here is:
+            # p1 = subprocess.Popen(["py-spy", "record", "-o", "profile.svg", "--", "python", "tests/objects.py"])
+            print(f"Running second profiler {command.split()}")
+        else:
+            print(
+                f"Second profiler {args.prof1} missing output flamegraph file location"
+            )
+
+    if args.prof2 is not None:
+        if args.prof1_output is not None:
+            command = args.prof2.replace("PID", str(p.pid))
+            p2 = subprocess.Popen(command.split())
+            print(f"Running third profiler {command}")
+        else:
+            print(
+                f"Third profiler {args.prof2} missing output flamegraph file location"
+            )
+
+    stderr, stdout = p.communicate()
```

Comment:
```suggestion
    stdout, stderr = p.communicate()
```
and fix usages below